drbert/data/dataset_readers.py

```diff
@@ -210,39 +297,191 @@ class RelationClassificationDatasetReader(DatasetReader):
+        return batch[:self.max_len_sentences_pair // 2]
+
+
+class STSDatasetReader(DatasetReader):
+    """Reads instances from a 3 columned TSV file(s) where each line is in the following format:
+
+    SENTENCE1   SENTENCE2    SIMILARITY_SCORE
+
+    E.g.,
+
+    Zyrtec 10 mg tablet 1 tablet...	cyclobenzaprine 5 mg tablet 1-2 tablets... 4
+
+    Example usage:
+        >>> from transformers import AutoTokenizer
+        >>> from dataset_readers import STSDatasetReader
+        >>> partitions={
+                'train': 'train_filename.tsv',
+                'test': 'test_filename.tsv',
+            }
+        >>> tokenizer = AutoTokenizer.from_pretrained('bert-base-cased')
+        >>> train_iter, test_iter = STSDatasetReader(
+                path='path/to/dataset', partitions=partitions, tokenizer=tokenizer,
+                batch_sizes=(16, 256)
+            ).textual_to_iterator()
+
+    Args:
+        path (str): Common prefix of the splits’ file paths.
+        partitions (dict): A dictionary keyed by partition ('train', 'validation', 'test')
+            containing the suffix to add to `path` for that partition.
+        tokenizer (transformers.PreTrainedTokenizer): The function used to tokenize strings into
+            sequential examples.
+        batch_sizes (tuple): Optional, tuple of batch sizes to use for the different splits, or None
+            to use the same batch_size for all splits. Defaults to None.
+        lower: (bool): Optional, True if text should be lower cased, False if not. Defaults to
+            False.
+        kwargs: Remaining keyword arguments. Will be ignored.
+    """
+    def __init__(self, path, partitions, tokenizer, batch_sizes=None, lower=False,
+                 device='cpu', **kwargs):
+        super(STSDatasetReader, self).__init__(
+            path=path, partitions=partitions, tokenizer=tokenizer, format='tsv', skip_header=False,
+            batch_sizes=batch_sizes, lower=lower,
+            # Sort examples according to length of sentence1 and sentence2
+            sort_key=lambda x: data.interleave_keys(len(x.sentence1), len(x.sentence2)),
+            device=device
+        )
+
+    def textual_to_iterator(self, **kwargs):
+        """Does whatever tokenization and/or processing is necessary to go from textual input to
+        iterators for training and evaluation for a NLI dataset.
+
+        Args:
+            kwargs: Passed to the `data.TabularDataset.splits` and
+            `data.BucketIterator.splits` class in the parent classes method `textual_to_iterator`.
+
+        Returns:
+            A tuple of `torchtext.data.iterator.BucketIterator` objects, one for each partition
+            is `self.partitions`.
+        """
+        # similarity scores are already continous numbers, no need to numericalize
+        self.LABEL.use_vocab = False
+        # cast similarity scores to floats (they are cast as strings)
+        self.LABEL.preprocessing = lambda x: float(x)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Lambda may not be necessary](https://app.codacy.com/app/JohnGiorgi/drbert/pullRequest?prid=4348433)